src/callback/file_callbacks.cpp

```diff
@@ -0,0 +1,167 @@
+#include "file_callbacks.h"
+#include <OS.hpp>
+
+namespace Callbacks{
+
+    GodotFileRunner* GodotFileRunner::get_singleton() {
```

Comment:
Why not to return `GodotFileRunner&` ? So that this becomes:
```cpp
    GodotFileRunner& GodotFileRunner::get_singleton() {
        static GodotFileRunner singleton;
        return singleton;
    }
```